rr_arduino/rr_arduino.ino

```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
     prev_time = millis();
   }
 }
+
+void raw_to_odom(double vel, int str_angle) {
+    long current_time = millis();
+    long time_diff = current_time - prev_time;
+    // If the robot is at the origin, calculate the position using steering angle and the velocity
+    if (str_angle != 1500 && x == 0.0 && y == 0.0) {
+        // Mapping str_angle integer to angle in degrees
+        theta = (str_angle - 1000) / (2000 - 1000) * 60 - 30;
+        x = vel * cos(theta * M_PI / 180.0) * time_diff;
```

Comment:
x is the integral so you would add the current value to what it was before

Reply:
I just set x and y to the calculation assuming the code will enter this if statement when the car hasn't moved yet (i.e. when the car is in the origin). In the else statement I do calculations using the kinematics equations. I think this should be okay?